### ISSUE TITLE
Architecture failure: hand-crafted weights should be ML-optimized — we are doing this backwards

### ISSUE THREAD

**Issue by holden** [razrfly/cinegraph#703]
## The Core Problem

We have been manually setting weights like "festival 30%, critics 20%, mob 30%" across five named profiles and then picking the best one. This is subjective guesswork dressed up as science. It is the exact problem ML is designed to solve.

**We have the training data. We have the labels. We should be fitting a model, not tuning knobs by hand.**

---

## What We Actually Have

A textbook supervised binary classification problem:

- **Features (X):** 5 computed criterion scores per movie — mob, critics, festival_recognition, cultural_impact, auteur_recognition (each 0–100)
- **Labels (y):** binary — is this film on the 1001 list or not?
- **Training data:** ~900 1001-list films + ~200,000 non-list films across 10 decades, all with computed scores already in the DB

We do not need to guess what the weights should be. A logistic regression will find them.

---

## What the Current Code Does Wrong

`compare_profiles` tests 5 hand-crafted weight combinations:

```
default:        mob 30%, critics 20%, festival 30%, cultural 15%, auteur 5%
festival-heavy: mob 10%, critics 10%, festival 60%, cultural 15%, auteur 5%
audience-first: mob 35%, critics 10%, festival 25%, cultural 25%, auteur 5%
critics-choice: mob 10%, critics 35%, festival 35%, cultural 15%, auteur 5%
auteur:         mob 15%, critics 15%, festival 30%, cultural 15%, auteur 25%
```

These five points are an infinitesimally small sample of a continuous 4-dimensional simplex (weights must sum to 1). We are picking the best of 5 arbitrary human guesses. There is nothing objective about this.

The current output ("default profile wins at 55.8%") tells us almost nothing. It tells us the best of 5 random human guesses.

---

### Step 1: Export the feature matrix

For every movie in every historical decade, compute the 5 criterion scores and the binary label:

```
movie_id | mob_score | critics_score | festival_score | cultural_score | auteur_score | on_1001_list
---------|-----------|---------------|----------------|----------------|--------------|-------------
  12345  |   78.2    |     65.0      |      45.0      |      32.1      |     60.0     |      1
  67890  |   55.0    |     40.0      |      90.0      |      12.0      |      0.0     |      0
```

### Step 2: Fit logistic regression

Logistic regression with no intercept and non-negative weights constrained to sum to 1 gives us exactly what we want: the weight of each feature in predicting list membership.

```python
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import StratifiedKFold

# Features are already 0-100 normalized
model = LogisticRegression(C=1.0, solver='lbfgs', max_iter=1000)
cv = StratifiedKFold(n_splits=5, shuffle=True, random_state=42)
scores = cross_val_score(model, X, y, cv=cv, scoring='accuracy')
```

The model coefficients are the optimal weights. No human opinion involved.

### Step 3: Cross-validate by decade, not randomly

Because older decades have less data coverage (sparse Metacritic, no festivals), we should use **decade-stratified cross-validation**: train on 8 decades, validate on 1, rotate. This tells us how well the model generalizes across eras, not just within them.

### Step 4: Run this for every target list separately

| Target | Training label | Expected weight differences |
|---|---|---|
| 1001 Movies | `canonical_sources ? '1001_movies'` | Heavy canonical + critics |
| Oscar Best Picture | Won Oscar for Best Picture | Heavy festival (AMPAS) |
| Cannes Palme d'Or | Won Palme | Heavy festival (Cannes) |
| Sight & Sound top 100 | `canonical_sources ? 'sight_sound_critics_2022'` | Different from 1001 entirely |

Each list gets its own model with its own optimal weights. The whole point of the architecture.

---

## What We Learn From This

Beyond just better weights, the model gives us:

1. **Feature importance** — which signal actually matters most for each list
2. **Era-specific models** — we can fit separate models for pre-1960 vs post-1960 if the optimal weights genuinely differ by era
3. **A principled accuracy ceiling** — if a well-fitted logistic regression tops out at 65%, that tells us we're missing features, not that our weights are wrong
4. **Confidence scores** — logistic regression gives probabilities, not just binary predictions, which is more useful for the prediction UI

---

## What This Exposes About the Current Architecture

The current `CriteriaScoring` module hard-codes 5 criterion weights and ships them as "named profiles." This should be replaced with:

1. A `ModelTrainer` that fits weights from historical data per target list
2. Stored model coefficients in the DB alongside the profile (so they can be retrained when new data arrives)
3. The "named profiles" become initial seeds for the optimizer, or are dropped entirely

The `HistoricalValidator` already computes accuracy by decade — it becomes the evaluation harness for comparing trained models against the hand-crafted baseline.

---

## Implementation Path

- [ ] Add a mix task `predictions.export_features` that dumps the feature matrix to CSV
- [ ] Fit logistic regression externally (Python/sklearn) or via Nx/Scholar in Elixir
- [ ] Compare ML-optimized weights vs current "default" profile — if accuracy doesn't improve significantly, we have a feature problem not a weight problem
- [ ] If accuracy improves, store optimal weights per target list in DB and expose in admin UI
- [ ] Retrain when canonical list data or festival data is updated

---

## Current baseline to beat

Overall accuracy with hand-tuned "default" profile: **55.8%**

If a logistic regression on the same features cannot meaningfully beat this, we need better features — not better weight-tuning. That distinction is the entire point.
